imageCompress/crop.py

```diff
@@ -1,18 +1,69 @@
+    image  = imread(img_path)   
+    height = image.shape[0]-H
+    width  = image.shape[1]-W
+    #---------------------------------------Exception Handling----------------------------------------------------#
+    # Exception handling for input validation like Type error, invalid values , unrealistic desired dimension     #
+    #-------------------------------------------------------------------------------------------------------------#
+    myError = False
+    if type(H) != int or type(W) !=int or type(image) is not np.ndarray:
+        myError = TypeError('Invalid Type')
+    elif H<0 or W <0 or H ==0 or W ==0 or H == int(-1e30) or W==int(-1e30):
+        myError = ValueError('Desired dimension should be positive')
+    elif H >= image.shape[0] + 1 or W >= image.shape[1] + 1:
+        myError = ValueError('Desired dimension should be less than original dimension')
+    if myError:
+        raise myError
+
+    #----------------------------------------Main Processing------------------------------------------------------#
+    # Removing rows from top and bottom by checking if the desired height is even or odd. For even height         #
+    # removing half number of rows of desired height from top and half number of rows of desired height from      #
+    # bottom. For Odd , dividing the height - 1 value and removing half number of rows from top , half + 1 number #
+    # row from bottom.Similar treatment for removing columns from left and right .                                #
+    #-------------------------------------------------------------------------------------------------------------#
+    if height % 2 ==0:                                     # For even desired height
+        start_row = int(height/2)
+        end_row   = int(image.shape[0]-height/2)
+    else:                                                  # For Odd desired height
+        start_row = int((height-1)/2)
+        end_row   = int((image.shape[0]-(height-1)/2)-1)
+
+    if width % 2 ==0:                                      # For even desired width
+        start_col = int(width/2)
+        end_col   = int((image.shape[1]-width/2))
+    else:                                                  # For Odd desired width
+        start_col = int((width-1)/2)
+        end_col   = int((image.shape[1]-(width-1)/2)-1)
+
+    img = image[start_row:end_row,start_col:end_col,:]     # Cropping image from all sides.
+    #-----------------------------------------Saving Cropped Image-----------------------------------------------#
+    # Saving the image and building the path for output                                                          #
+    #------------------------------------------------------------------------------------------------------------#
+    path = os.path.dirname(os.path.abspath(img_path))
+    crop_img_path = os.path.join(path, "crop_img.png")
+    imsave(crop_img_path, img)
+
+    return(crop_img_path)
```

Comment:
This seems like an R return syntax. In python, we can simply use `return crop_img_path`.

Reply:
Done